src/org/rascalmpl/parser/uptr/recovery/ToTokenRecoverer.java

```diff
@@ -51,13 +52,36 @@ public class ToTokenRecoverer implements IRecoverer<IConstructor> {
 	private URI uri;
 	private IdDispenser stackNodeIdDispenser;
 	private ExpectsProvider<IConstructor> expectsProvider;
+	private int maxRecoveryAttempts;
+	private int maxRecoveryTokens;
 
 	private Set<Long> processedNodes = new HashSet<>();
 
-	public ToTokenRecoverer(URI uri, ExpectsProvider<IConstructor> expectsProvider, IdDispenser stackNodeIdDispenser) {
+	private static final int DEFAULT_SKIP_WINDOW = 2048;
+	private static int skipWindow = readEnvVar("RASCAL_RECOVERER_SKIP_WINDOW", DEFAULT_SKIP_WINDOW);
+
+	private int count = 0;
+
+	private static int readEnvVar(String envVar, int defaultValue) {
+		String limitSpec = System.getenv(envVar);
+		if (limitSpec != null) {
+			try {
+				System.err.println("Using enviroment variable " + envVar + " = " + limitSpec);
+				return Integer.parseInt(limitSpec);
+			} catch (NumberFormatException e) {
+				return defaultValue;
+			}
+		}
+		System.err.println("Using default value for " + envVar + " = " + defaultValue);
```

Comment:
we don't want to be printing to stderr all the time. this can break the REPL.

Reply:
This code is needed while experimenting but will be removed (or maybe hidden behind a guard) in the final version of this PR.